pkg/app2/appserver/proc.go

```diff
@@ -0,0 +1,99 @@
+package appserver
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/skycoin/skycoin/src/util/logging"
+)
+
+// Proc is a wrapper for a skywire app. Encapsulates
+// the running proccess itself and the RPC server for
+// app/visor communication.
+type Proc struct {
+	key    Key
+	config Config
+	log    *logging.Logger
+	rpcS   *Server
+	cmd    *exec.Cmd
+}
+
+// NewProc constructs `Proc`.
+func NewProc(log *logging.Logger, c Config, args []string) (*Proc, error) {
+	key := GenerateAppKey()
+
+	binaryPath := getBinaryPath(c.BinaryDir, c.Name, c.Version)
+
+	const (
+		appKeyEnvFormat   = "APP_KEY=%s"
+		sockFileEnvFormat = "SW_UNIX=%s"
+	)
```

Comment:
We also need to provide the following via envs:
- Local public key of visor.
- `ProcID`? Are we still using that?

Reply:
@evanlinjin no, there's no need of `ProcID`, I'm removing it where it's left unused. App key is for authentication now

Reply:
Done